docs/content/docs/concepts/glossary.md

```diff
@@ -25,182 +25,605 @@ under the License.
+  getters and setters
+  
+Flink analyzes the structure of POJO types and can process POJOs more efficiently than general types.
+
+#### Process Functions
+
+This type of function combines event processing with timers and state and is the basis for creating 
+event-driven applications with Flink.
+
+#### Processing Time
+
+The time when a specific operator in your pipeline is processing the event. Computing analytics based 
+on processing time can cause inconsistencies and make it difficult to re-analyze historic data or test 
+new implementations.
+
+#### Queryable State 
+
+This is managed keyed (partitioned) state that can be accessed from outside of Flink during runtime.
+
+#### Raw State
+
+This is state that operators keep in their own data structures. When checkpointed, only a sequence of 
+bytes is written into the checkpoint and Flink knows nothing about the state’s data structures and will 
+see only the raw bytes.
+
+[Keyed state](#keyed-state) and [operator state](#operator-state) exist in two forms: [managed](#managed-state) and raw.
 
 #### Record
 
-Records are the constituent elements of a data set or data stream. [Operators](#operator) and
-[Functions](#Function) receive records as input and emit records as output.
+Records are the elements that make up a [DataStream](#datastream). [Operators](#operator) and [functions](#function) 
+receive records as input and emit records as output.
+
+#### ResourceManager
+
+This is part of the [JobManager](#JobManager) and is responsible for resource de-/allocation and 
+provisioning in a Flink cluster.
+
+#### Rich Functions
+
+A RichFunction is a "rich" variant of Flink's function interfaces for data transformation. These functions 
+have some additional methods needed for working with managed keyed state such as `open(Configuration c)`, 
+`close()`, `getRuntimeContext()`.
+
+#### Rolling Total
+
+The sum of a sequence of numbers which is updated each time a new number is added to the sequence, 
+by adding the value of the new number to the previous rolling total.
 
 #### (Runtime) Execution Mode
 
-DataStream API programs can be executed in one of two execution modes: `BATCH`
-or `STREAMING`. See [Execution Mode]({{< ref "/docs/dev/datastream/execution_mode" >}}) for more details.
+DataStream API programs can be executed in one of two execution modes: `BATCH` or `STREAMING`. 
+See [Execution Mode]({{< ref "/docs/dev/datastream/execution_mode" >}}) for more details.
+
+#### Savepoint
+
+A [snapshot](#snapshot) triggered manually by a user (or an API call) for some operational purpose, 
+such as a stateful redeploy/upgrade/rescaling. Savepoints are always complete and aligned and are 
+optimized for operational flexibility.
+
+#### Scalar
+
+A scalar refers to a single value. This is in contrast to a set of values. 
+
+#### Schema
+
+This refers to the organization or structure of data as a blueprint. 
```

Comment:
I would remove this one, this is an informatics known concept and there is nothing specific to add here in relation to Flink